### PR TITLE
Add React Goods selector

### PR DESCRIPTION
**Pull request by Ruslan-Pashkov**
[DEMO LINK](https://ruslan-pashkov.github.io/react_goods-selector/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -15,71 +17,82 @@ export const goods = [
...
+                  >
+                    <td>
+                      <button
+                        data-cy={isSelected ? 'RemoveButton' : 'AddButton'}
+                        type="button"
+                        className={cn({
+                          button: true,
+                          'is-info': isSelected,
+                        })}
+                        onClick={() => this.selectItem(good)}
```
Comment (Anastasia4sik) [inline src/App.tsx:80]: Checklist p.3

File: src/App.tsx [hunk 2]
```diff
@@ -15,71 +17,82 @@ export const goods = [
...
+          selectedGood
+            ? (
+              <h1 className="title is-flex is-align-items-center">
+                {`${selectedGood} is selected`}
+                {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+                <button
+                  data-cy="ClearButton"
+                  type="button"
+                  className="delete ml-3"
+                  onClick={() => this.setState({ selectedGood: '' })}
```
Comment (Anastasia4sik) [inline src/App.tsx:51]: Checklist p.3

**Review by Anastasia4sik — CHANGES_REQUESTED** [review]
Fix a few things

File: src/App.tsx [hunk 3]
```diff
@@ -15,71 +17,82 @@ export const goods = [
...
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  selectItem = (item: string) => {
+    const condition = item === this.state.selectedGood;
```
Comment (Ihor-Karpyn) [inline src/App.tsx:30]: Let's think more about variable name

File: src/App.tsx [hunk 4]
```diff
@@ -15,71 +17,82 @@ export const goods = [
...
+    const valueToSet = condition ? '' : item;
```
Comment (Ihor-Karpyn) [inline src/App.tsx:31]: I hope this variable redundant
```suggestion
```

File: src/App.tsx [hunk 5]
```diff
@@ -15,71 +17,82 @@ export const goods = [
...
-            </button>
-          </td>
+    return (
+      <main className="section container">
+        {
+          selectedGood
+            ? (
+              <h1 className="title is-flex is-align-items-center">
+                {`${selectedGood} is selected`}
+                {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
```
Comment (Ihor-Karpyn) [inline src/App.tsx:46]: Let's add a label, the linter doesn't exist to disable it

File: src/App.tsx [hunk 6]
```diff
@@ -15,71 +17,82 @@ export const goods = [
...
+        }
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+        <table className="table">
+          <tbody>
+            {
+              goods.map(good => {
```
Comment (Ihor-Karpyn) [inline src/App.tsx:61]:
```suggestion
            {goods.map(good => {
```

**Review by Ihor-Karpyn — CHANGES_REQUESTED** [review]
Almost done

**Review by Ihor-Karpyn — APPROVED** [review]

**Pull request closed without merging**